src/OrchardCore.Modules/OrchardCore.AutoSetup/Startup.cs

```diff
@@ -0,0 +1,109 @@
+using OrchardCore.Environment.Shell.Models;
+
+namespace OrchardCore.AutoSetup
+{
+    /// <summary>
+    /// The AutoSetup feature startup.
+    /// </summary>
+    public sealed class Startup : StartupBase
+    {
+        /// <summary>
+        /// The Shell/Tenant configuration.
+        /// </summary>
+        private readonly IShellConfiguration _shellConfiguration;
+
+        /// <summary>
+        /// AutoSetup Configuration Section Name 
+        /// </summary>
+        private const string ConfigSectionName = "OrchardCore_AutoSetup";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Startup"/> class.
+        /// </summary>
+        /// <param name="shellConfiguration">
+        /// The shell configuration.
+        /// </param>
+        public Startup(IShellConfiguration shellConfiguration)
+        {
+            this._shellConfiguration = shellConfiguration;
+        }
+
+        /// <summary>
+        /// The configure services.
+        /// </summary>
+        /// <param name="services">
+        /// The services.
+        /// </param>
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            var configuration = _shellConfiguration.GetSection(ConfigSectionName);
+            services.Configure<AutoSetupOptions>(configuration);
+        }
+
+        /// <summary>
+        /// Configure application pipeline.
+        /// </summary>
+        /// <param name="app">
+        /// The app.
+        /// </param>
+        /// <param name="routes">
+        /// The routes.
+        /// </param>
+        /// <param name="serviceProvider">
+        /// The "Shell Scope" service provider.
+        /// </param>
+        public override void Configure(IApplicationBuilder app, IEndpointRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var currentShellSettings = serviceProvider.GetRequiredService<ShellSettings>();
+            var logger = serviceProvider.GetRequiredService<ILogger<Startup>>();
+
+
+            if (currentShellSettings.State == TenantState.Uninitialized)
+            {
+                var optionsAccessor = serviceProvider.GetRequiredService<IOptions<AutoSetupOptions>>();
+                var options = optionsAccessor.Value;
+
+                if (options != null)
+                {
+                    var validationContext = new ValidationContext(options, serviceProvider, null);
+                    var validationErrors = options.Validate(validationContext);
+
+                    if (validationErrors.Any())
+                    {
+                        var stringBuilder = new StringBuilder();
+                        foreach (var error in validationErrors)
+                        {
+                            stringBuilder.Append(error.ErrorMessage);
+                        }
+
+                        logger.LogError("AutoSetup did not start, configuration has following errors: {errors}", stringBuilder.ToString());
```

Comment:
Here, you are using localized strings for the validation, but then recording the validation errors in the log file, so it doesn't make sense that they are localized. Because log files don't contain localization.

If it was going to the front end, it would make sense for them to be localized, but when they are being appended to the log file, then it makes less sense. Because none of the ASP.Net errors, are localized, nor is the actual error message, of `AutoSetup did not start`.

So having `AutoSetup did not start : {errors in another language}` doesn't quite add up to me?

Reply:
thanks for your review, I agree with you, I've removed localization